core/src/main/java/com/linecorp/armeria/server/AbstractServiceBuilder.java

```diff
@@ -0,0 +1,124 @@
+/*
+ * Copyright 2019 LINE Corporation
+ *
+ * LINE Corporation licenses this file to you under the Apache License,
+ * version 2.0 (the "License"); you may not use this file except in compliance
+ * with the License. You may obtain a copy of the License at:
+ *
+ *   https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ */
+
+package com.linecorp.armeria.server;
+
+import static java.util.Objects.requireNonNull;
+
+import java.nio.charset.Charset;
+import java.time.Duration;
+import java.util.function.Function;
+
+import com.linecorp.armeria.common.HttpRequest;
+import com.linecorp.armeria.common.HttpResponse;
+import com.linecorp.armeria.common.logging.ContentPreviewerFactory;
+import com.linecorp.armeria.server.logging.AccessLogWriter;
+
+/**
+ * A builder that implements {@link ServerBuilder} by delegating all calls
+ * to {@link DefaultServiceBuilder}
+ */
+abstract class AbstractServiceBuilder implements ServiceBuilder {
```

Comment:
Could you explain why we need this class? It seems to me that we can just extend `DefaultServiceBuilder` and remove this class, no?

Reply:
There is no much difference except `AbstractServiceConfigSetters`(which was AbstractServiceBuilder) provides `build0()` that builds `ServiceConfigBuilder` and calls the abstract method `serviceConfigBuilder` for sub-classes to do something with `ServiceConfigBuilder`.

`AnnotatedServiceBindingBuilder`'s `buildAnnotated` will look like this:

```
final ServiceConfigBuilder serviceConfigBuilder = toServiceConfigBuilder(e.route(), s);
serverBuilder.serviceConfigBuilder(serviceConfigBuilder);
```

instead of 

`build0(e.route(), s);`

I can remove this class if this class is accomplishing little/no value.

Reply:
> I can remove this class if this class is accomplishing little/no value.

Then, let's remove this class and use `DefaultServiceConfigSetters`. :-)